integration/src/integration/java/pl/allegro/tech/hermes/integration/HermesClientPublishingTest.java

```diff
@@ -47,6 +51,38 @@ public void shouldPublishUsingRestTemplate() {
         runTestSuiteForHermesClient(client);
     }
 
+    @Test
+    public void shouldPublishUsingOkHttpClient() {
+        // given
+        HermesClient client = hermesClient(new OkHttpSender()).withURI(topicURI).build();
+
+        // when & then
+        runTestSuiteForHermesClient(client);
+    }
+
+    @Test
+    public void shouldPublishUsingOkHttpClientUsingSSL() {
+        // given
+        SSLContextSupplier sslSupplier = getSslContextSupplier();
+        HermesClient client = hermesClient(new OkHttpSender(sslSupplier.get())).withURI(topicURI_SSL).build();
+
+        // when & then
+        runTestSuiteForHermesClient(client);
+    }
+
+    @Test
+    public void shouldCommunicateWithHermesUsingHttp2() {
+        // given
+        SSLContextSupplier sslSupplier = getSslContextSupplier();
```

Comment:
do we need `SSLContextSupplier` here? `SSLContextSupplier#get` is called anyways (in both tests), private method could just call get and return it